File: AspNetMvcUnleashed/0672329980_source/Chapter18/VB/UnleashedBlog/UnleashedBlog/Controllers/BlogController.vb
Public Class BlogController
    Inherits System.Web.Mvc.Controller

    Private _repository As BlogRepositoryBase

    Public Sub New()
        Me.New(New EntityFrameworkBlogRepository())
    End Sub


    Public Sub New(ByVal repository As BlogRepositoryBase)
        _repository = repository
    End Sub


    Public Function Index() As ActionResult
        Return View(_repository.ListBlogEntries())
    End Function


    Public Function Create()
        Return View()
    End Function

    <AcceptVerbs(HttpVerbs.Post)> _
    Public Function Create(ByVal blogEntryToCreate As BlogEntry) As ActionResult
        _repository.CreateBlogEntry(blogEntryToCreate)

        Return RedirectToAction("Index")
    End Function

End Class

File: AspNetMvcUnleashed/0672329980_source/Chapter18/CS/UnleashedBlog/UnleashedBlog/Controllers/BlogController.cs
using System.Collections.Generic;
using System.Web.Mvc;
using UnleashedBlog.Models;
using UnleashedBlog.Models.EntityFramework;

namespace UnleashedBlog.Controllers
{
    public class BlogController : Controller
    {
        private BlogRepositoryBase _repository;


        public BlogController()
            : this(new EntityFrameworkBlogRepository()) { }

        public BlogController(BlogRepositoryBase repository)
        {
            _repository = repository;
        }

        public ActionResult Index()
        {
            return View(_repository.ListBlogEntries());
        }


        public ActionResult Create()
        {
            return View();
        }

        [AcceptVerbs(HttpVerbs.Post)]
        public ActionResult Create(BlogEntry blogEntryToCreate)
        {
            _repository.CreateBlogEntry(blogEntryToCreate);
            return RedirectToAction("Index");
        }
    }
}
